packages/wrangler/src/config/index.ts

```diff
@@ -91,33 +91,12 @@ export function readConfig(
 	requirePagesConfig?: boolean,
 	hideWarnings: boolean = false
 ): Config {
-	let rawConfig: RawConfig = {};
-
-	if (!configPath) {
-		configPath = findWranglerConfig(process.cwd());
-	}
+	const { rawConfig, configPath: updateConfigPath } = readRawConfig(
+		configPath,
+		requirePagesConfig
+	);
 
-	try {
-		// Load the configuration from disk if available
-		if (configPath?.endsWith("toml")) {
-			rawConfig = parseTOML(readFileSync(configPath), configPath);
-		} else if (configPath?.endsWith("json") || configPath?.endsWith("jsonc")) {
-			rawConfig = parseJSONC(readFileSync(configPath), configPath);
-		}
-	} catch (e) {
-		// Swallow parsing errors if we require a pages config file.
-		// At this point, we can't tell if the user intended to provide a Pages config file (and so should see the parsing error) or not (and so shouldn't).
-		// We err on the side of swallowing the error so as to not break existing projects
-		if (requirePagesConfig) {
-			logger.error(e);
-			throw new FatalError(
-				`Your ${configFileName(configPath)} file is not a valid Pages config file`,
-				EXIT_CODE_INVALID_PAGES_CONFIG
-			);
-		} else {
-			throw e;
-		}
-	}
+	configPath = updateConfigPath;
```

Comment:
nit, but can we call the return from `readRawConfig()` `resolvedConfigPath` or something like that and avoid this reassigning?

Reply:
cleaned up + moved resolving out (not sure what i was thinking...) https://github.com/cloudflare/workers-sdk/pull/7291/files#diff-80a0404f59f781e58a6e6892e5457ca6bd1c7b7e5687155159bde57f6cc65452